src/auto/dbus_proxy.rs

```diff
@@ -0,0 +1,927 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use gio_sys;
+use glib;
+use glib::object::Cast;
+use glib::object::IsA;
+use glib::signal::connect_raw;
+use glib::signal::SignalHandlerId;
+use glib::translate::*;
+use glib::GString;
+use glib::StaticType;
+use glib::Value;
+use glib_sys;
+use gobject_sys;
+use std::boxed::Box as Box_;
+use std::fmt;
+use std::mem::transmute;
+use std::pin::Pin;
+use std::ptr;
+use BusType;
+use Cancellable;
+use DBusCallFlags;
+use DBusConnection;
+use DBusInterface;
+use DBusInterfaceInfo;
+use DBusProxyFlags;
+#[cfg(any(unix, feature = "dox"))]
+use UnixFDList;
+
+glib_wrapper! {
+    pub struct DBusProxy(Object<gio_sys::GDBusProxy, gio_sys::GDBusProxyClass, DBusProxyClass>) @implements DBusInterface;
+
+    match fn {
+        get_type => || gio_sys::g_dbus_proxy_get_type(),
+    }
+}
+
+impl DBusProxy {
+    pub fn new_for_bus_sync<P: IsA<Cancellable>>(
+        bus_type: BusType,
+        flags: DBusProxyFlags,
+        info: Option<&DBusInterfaceInfo>,
+        name: &str,
+        object_path: &str,
+        interface_name: &str,
+        cancellable: Option<&P>,
+    ) -> Result<DBusProxy, glib::Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = gio_sys::g_dbus_proxy_new_for_bus_sync(
+                bus_type.to_glib(),
+                flags.to_glib(),
+                info.to_glib_none().0,
+                name.to_glib_none().0,
+                object_path.to_glib_none().0,
+                interface_name.to_glib_none().0,
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                &mut error,
+            );
+            if error.is_null() {
+                Ok(from_glib_full(ret))
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    pub fn new_sync<P: IsA<Cancellable>>(
```

Comment:
Just a note: I still don't get why there is both `_sync` and the "normal" one. It's weird to see that in the GIO API...

Reply:
Lots of stuff in this API has sync and async versions, because.. reasons?

("normal" is async)

Reply:
No I mean, there is "normal", "async" and "not_async". One of the three is very likely unneeded. :)

Reply:
The DBus API has `_sync` (blocking) and without suffix (async function). It's just the opposite of elsewhere for reasons.